### PR TITLE
Remove warnings emitted during GH actions

### PR DESCRIPTION
**Pull request by konstntokas**
For this PR, I went through the warnings emitted during GH actions and resolved those which were easy to fix. I ignored the `DeprecationWarnings` emitted by `xcube` itself. In total the number of warnings could be decreased from ~10500 to ~7700. 

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by konstntokas** [comment]
In `CHANGES.md`, a second section for Other changes was opened in [one of the last PR](https://github.com/xcube-dev/xcube/pull/986/files#diff-d975bf659606195d2165918f93e1cf680ef68ea3c9cab994f033705fea8238b2) in Changes in 1.6.0 (in development). I merge them together with this PR.

File: xcube/core/gridmapping/base.py [hunk 1]
```diff
@@ -172,15 +172,25 @@ def derive(
                 with self._lock:
                     if other._xy_coords is not None:
                         other._xy_coords = other._xy_coords.chunk(
-                            other.xy_coords_chunks
+                            {
+                                dim: size for (dim, size) in zip(
+                                    other._xy_coords.dims, other.xy_coords_chunks
+                                    )
+                            }
```
Comment (forman) [inline xcube/core/gridmapping/base.py:179]: Use black

File: xcube/core/gridmapping/base.py [hunk 2]
```diff
@@ -172,15 +172,25 @@ def derive(
...
             if other._xy_coords is not None:
                 other._xy_coords = other._xy_coords[:, ::-1, :]
-                other._xy_coords = other._xy_coords.chunk(other.xy_coords_chunks)
+                other._xy_coords = other._xy_coords.chunk(
+                    {
+                        dim: size for (dim, size) in zip(
+                            other._xy_coords.dims, other.xy_coords_chunks
+                            )
+                    }
+                )
```
Comment (forman) [inline xcube/core/gridmapping/base.py:193]: Use black

**Review by forman — COMMENTED** [review]
Wonderful. Just two formatting issue, but this can be merged still.

**Pull request merged**